### PR TITLE
feat(providers): add Pipedrive provider

### PR DESCRIPTION
**Pull request by estevanjantsk**
## Reasoning 💡

Added Pipedrive as a provider, because the lib does not have yet.
I tested for both version v3 and v4 and it's working.

## Checklist 🧢

- [x] Documentation [[PR](https://github.com/nextauthjs/docs/pull/83)]
- [ ] Tests
- [ ] Ready to be merged

### CODE REVIEW

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thank you for this PR!

I left some comments. I do not wish to "backport" this to v3, so the v3 related configuration is unnecessary. Let us focus on getting this working in v4 for now. Could you address my concerns and let me know if it works for you?

By the way, are you interested in adding this as TypeScript? You can see any of the `.ts` files in this directory for inspiration: https://github.com/nextauthjs/next-auth/tree/beta/src/providers

File: src/providers/pipedrive.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  company_domain?:        string;
+  company_country?:       string;
+  company_industry?:      string;
+  language?:              Language;
+}
+
+interface Language {
+  language_code?: string;
+  country_code?:  string;
+}
```
Comment (balazsorban44) [inline src/providers/pipedrive.ts:40]: Could you have this as a single interface? Also, if you have it configured, could you run Prettier on the file for consistent formatting with the rest of the codebase? :pray:

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

File: src/providers/pipedrive.ts (hunk 1 above)
Comment (estevanjantsk) [reply src/providers/pipedrive.ts]: I've fixed the interface. however, I don't have Prettier configured, could you run for me, please?

Comment (lluia) [reply src/providers/pipedrive.ts]: @estevanjantsk I run `prettier` for you on the file 👍🏽 , for next time, if you already did `npm install` on the cloned project just run `npm run lint:fix` and it'll format the file for you 💅🏽

I think this provider is ready to go then @balazsorban44  🤞🏽 ?

**Review by balazsorban44 — APPROVED** [review]
Thanks! I will take your word that you have tested, but as a proof, could you just link to an image or the JSON returned by the `profile` callback that we can refer to later on if someone has problems with this provider in the future?

Something like here: https://github.com/nextauthjs/next-auth/pull/2848#issue-1008276523 or just the JSON output.

Thank you!

**Pull request merged**